Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/TotScatCalculateSelfScattering.py

```diff
@@ -79,28 +79,42 @@ def PyExec(self):
                                           BinningForFit=[min_x, 10 * width_x, max_x],
                                           FitSpectrumWith="CubicSpline")
         self_scattering_correction = CalculatePlaczekSelfScattering(InputWorkspace=raw_ws,
-                                                                    IncidentSpecta=fit_spectra,
-                                                                    CrystalDensity=crystal_density)
+                                                                    IncidentSpectra=fit_spectra,
+                                                                    CrystalDensity=crystal_density,
+                                                                    Version=1)
+        # Convert to Q
+        self_scattering_correction = ConvertUnits(InputWorkspace=self_scattering_correction,
+                                                  Target="MomentumTransfer", EMode='Elastic')
         cal_workspace = LoadCalFile(InputWorkspace=self_scattering_correction,
                                     CalFileName=cal_file_name,
                                     Workspacename='cal_workspace',
                                     MakeOffsetsWorkspace=False,
-                                    MakeMaskWorkspace=False)
-        self_scattering_correction = DiffractionFocussing(InputWorkspace=self_scattering_correction,
-                                                          GroupingFilename=cal_file_name)
+                                    MakeMaskWorkspace=False,
+                                    MakeGroupingWorkspace=True)
+        ssc_min_x, ssc_max_x = float('inf'), float('-inf')
```

Comment:
This calculation for the rebin parameters looks fine now. If we were pursuing this approach of focussing the correction I think it would be better if we calculated different rebin parameters per group and then called GroupDetectors per group. DiffractionFocussing calculates rebin parameters per group (but with the unwanted side effect that it always uses log bins). Not sure it's worth you doing this work now though. So please could you add a comment in here saying "TO DO: calculate rebin parameters per group" or similar

I noticed this because the q range of the correction for bank 1 is quite small (0 to ~30 only) so when you rebin using a common set of rebin parameters (spanning a q range of ~0-375) across the whole workspace you end up with quite a lot of zeroes for the bank 1 correction for the high q values (in daniel_ssc). The focussed correction eventually gets rebinned to match the focused data in polaris_algs so doesn't ultimately have a huge effect but perhaps reduces the granularity of the correction for bank 1 unnecessarily